site/content/docs/main/csi-snapshot-data-movement.md

```diff
@@ -0,0 +1,390 @@
+
+## Install Velero with CSI support on source cluster
+
+On source cluster, Velero needs manipulate CSI snapshots through the CSI volume snapshot APIs, so you must enable the `EnableCSI` feature flag and install the Velero [CSI plugin][2] on the Velero server.  
+
+Both of these can be added with the `velero install` command.
+
+```bash
+velero install \
+--features=EnableCSI \
+--plugins=<object storage plugin>,velero/velero-plugin-for-csi:v0.6.0 \
+...
+```
+
+### Configure storage class on target cluster
+
+For Velero built-in data movement, CSI facilities are not required necessarily in the target cluster. On the other hand, the volume provision should work similarly to the source cluster. For example, the same storage class should be working in the target cluster.  
+By default, Velero won't restore storage class resources from the backup since they are cluster scope resources. However, if you specify the `--include-cluster-resources` restore flag, they will be restored. For a cross provider scenario, the storage class from the source cluster is probably not usable in the target cluster.  
+In either of the above cases, the best practice is to create a working storage class in the target cluster with the same name as it in the source cluster. In this way, even though `--include-cluster-resources` is specified, Velero restore will skip restoring the storage class since it finds an existing one.  
+Otherwise, if the storage class name in the target cluster is different, you can change the PVC's storage class name during restore by the [changing PV/PVC storage class][10] method. You can also configure to skip restoring the storage class resources from the backup since they are not usable.  
+
+### Customized Data Movers
+
+If you are using a customized data mover, follow the data mover's instructions for any further prerequisites.  
+For Velero side configurations mentioned above, the installation and configuration of node-agent may not be required.  
+
+
+## To back up
+
+Velero uses a new custom resource `DataUpload` to drive the data movement. The selected data mover will watch and reconcile the CRs.  
+Velero allows users to decide whether the CSI snapshot data should be moved per backup.  
+Velero also allows users to select the data mover to move the CSI snapshot data per backup.  
+The both selections are simply done by a parameter when running the backup.  
+
+To take a backup with Velero's built-in data mover:
+
+```bash
+velero backup create NAME --snapshot-move-data OPTIONS...
+```
+
+Or if you want to use a customized data mover:
+```bash
+velero backup create NAME --snapshot-move-data --data-mover DATA-MOVER-NAME OPTIONS...
+```
+
+When the backup starts, you will see the `VolumeSnapshot` and `VolumeSnapshotContent` objects created, but after the backup finishes, the objects will disppear.  
+After snapshots are created, you will see one or more `DataUpload` CRs created.  
```

Comment:
Do we want to mention that VBDM will be creating a DataUpload CR per PVC/CSI snapshot included in the backup ?

Reply:
It has been mentioned in line 299:
`DataUpload` - represents a data upload of a volume snapshot. The CSI plugin creates one `DataUpload` per CSI snapshot.